### PR TITLE
Resets windows sleep timer on pad input

### PR DESCRIPTION
**Pull request by RipleyTom**

### CODE REVIEW

**Comment by yiweigang** [comment]
I think it must be have a option setting

**Comment by hcorion** [comment]
Why would you want your computer to go to sleep while playing a game? We don't need an option for this, it would just clutter up the gui and would never get used.

**Comment by paulsapps** [comment]
Because if you go AFK why wouldn't you want it to sleep? Otherwise your computer just burns power running an idle game..

If you are actually not AFK then the machine should never sleep since there is user input.

**Comment by RipleyTom** [comment]
Gamepad use doesn't stop windows going to sleep, that's why.

**Comment by Blaypeg** [comment]
I believe any application should follow the OS settings for power management and have a setting if it will override this.

**Comment by yiweigang** [comment]
maybe you will goto sleep without closing the game

**Comment by yiweigang** [comment]
I have a idea,If the emulator window is normal size or maxsize No sleep,else let windows go sleep

**Comment by RipleyTom** [comment]
I changed it to only reset timer on pad input.
It should satisfy everyone, hopefully.

**Comment by yiweigang** [comment]
thank @RipleyTom for your change

**Comment by twdarkeh** [comment]
Does this work on DS4 as well?

**Comment by yiweigang** [comment]
just xinput @twdarkeh

**Comment by RipleyTom** [comment]
Added timer reset on ds4 and refactored

File: rpcs3/ds4_pad_handler.cpp [hunk 1]
```diff
@@ -394,6 +398,21 @@ void ds4_pad_handler::ProcessData()
...
 		}
 
+#ifdef _WIN32
+		bool any_button_pressed = false;
+
+		for (int i = 6; i < 16; i++)
+		{
+			if (pad.m_buttons[i].m_pressed)
+			{
+				any_button_pressed = true;
```
Comment (Blaypeg) [inline rpcs3/ds4_pad_handler.cpp]: Could you move the SetThreadExecutionState command here and remove need for the any_button_pressed variable?

**Comment by RipleyTom** [comment]
You're right, fixed.

**Comment by MSuih** [comment]
With this implementation you might run into issues where the screen turns off during a long cutscene or end credits. It's not a huge deal though since most games do not have 5+ minutes of cutscenes without some sort of interaction and end credits aren't that important, but I thought it was still worth mentioning.

**Pull request merged**